src/vehicle.cpp

```diff
@@ -3044,18 +3049,18 @@ int vehicle::total_power(bool const fueled) const
     return pwr;
 }
 
-int vehicle::acceleration(bool const fueled) const
+float vehicle::acceleration(bool const fueled) const
 {
     if( ( engine_on && has_engine_type_not( fuel_type_muscle, true ) ) || skidding ) {
-        return safe_velocity( fueled ) * k_mass() / ( 1 + strain () ) / 10;
+        return safe_velocity( fueled ) * k_mass() / ( 1.0f + strain () ) / 10.0f;
 
     } else if ((has_engine_type(fuel_type_muscle, true))){
         //limit vehicle weight for muscle engines
-        const units::mass mass = total_mass() / 1000;
+        const units::mass mass = total_mass() / 1000.0f;
         ///\EFFECT_STR caps vehicle weight for muscle engines
-        const units::mass move_mass = std::max( g->u.str_cur * 25_gram, 150_gram ) * 1000;
+        const units::mass move_mass = std::max( g->u.str_cur * 25_gram, 150_gram ) * 1000.0f;
```

Comment:
Btw. the multiplication with a floating point instead of an integer is pointless. It creates a temporary `quantity<float, mass_in_gram_tag>` instance that is assigned to a `units::mass`, which is `int`-based, so the floating point value is converted back to an integer.

Reply:
Didn't realize units::mass had to be an integer. Thanks!